tools/generate-js.py

```diff
@@ -102,17 +102,66 @@ def generate_tlvtype(tlvtype: 'TlvMessageType'):
     # Now, generate table
     print('const tlv_{} = {{\n'.format(tlvtype.name))
     for f in tlvtype.fields:
-        print('    {num}: [ "{fname}", towire_{tlvname}_{fname}, fromwire_{tlvname}_{fname} ],\n'
+        print('    {num}: [ "{fname}", towire_{tlvname}_{fname}, fromwire_{tlvname}_{fname} ],'
               .format(num=f.number, tlvname=tlvtype.name, fname=f.name))
     print('}\n')
 
 
 def generate_msgtype(name: str):
-    raise RuntimeError("FIXME: Implement!")
-
+    print('function towire_{tlvname}(value)\n'
+              '{{'
+              .format(tlvname=name.name))
+    print('    let _n = 0;\n'
+          '    let buf = Buffer;')
+    for f in name.fields:
+        generate_towire_field(f, name.fields)
+    print('    assert(value.length() == _n);\n'
+          '    return buf;\n'
+          '}')
```

Comment:
Similarly, fromwire for messages should:
1. call fromwire_u16 to get the type.
2. if it's not correct, return (whatever you think appropriate on failure?).  Or maybe throw an exception.